datafusion/expr/src/function.rs

```diff
@@ -218,7 +218,9 @@ pub fn return_type(
         BuiltinScalarFunction::Concat => Ok(DataType::Utf8),
         BuiltinScalarFunction::ConcatWithSeparator => Ok(DataType::Utf8),
         BuiltinScalarFunction::DatePart => Ok(DataType::Float64),
-        BuiltinScalarFunction::DateTrunc | BuiltinScalarFunction::DateBin => {
+        // DateTrunc always makes nanosecond timestamps
+        BuiltinScalarFunction::DateTrunc => Ok(DataType::Timestamp(TimeUnit::Nanosecond, None)),
```

Comment:
here is the actual bugfix 

I think it is a reasonable question if date_trunc should be changed to return the same type as its input, but the current implementation always creates nanosecond timestamps;

https://github.com/alamb/arrow-datafusion/blob/main/datafusion/physical-expr/src/datetime_expressions.rs#L266